### PR TITLE
[TASK] Release 2.2.0

### PR DESCRIPTION
**Pull request by t3easy**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
No changes in PHP files beside the version string

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
New release

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -35,5 +34,10 @@
     },
     "bin": [
         "surf"
-    ]
+    ],
+    "config": {
+        "platform": {
+            "php": "7.0.33"
```
Comment (simonschaufi) [inline composer.json:40]: why?

Comment (t3easy) [reply composer.json:40]: Because this prevents locking dependencies that require higher PHP versions during `composer update` if you have e.g. PHP 7.2 that executes composer.

Comment (t3easy) [reply composer.json:40]: 7.0.33 is the latest version in Debian oldstable https://packages.debian.org/stretch/php7.0-cli. I think we should not be more legacy than that.

Comment (simonschaufi) [reply composer.json:40]: ok

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -39,6 +40,8 @@ jobs:
         php-version: ${{ matrix.php }}
         coverage: none # disable xdebug, pcov
     - run: composer install --no-suggest --no-progress
+    - run: composer update --no-suggest --no-progress
```
Comment (simonschaufi) [inline .github/workflows/tests.yaml:43]: that will make the tests very very slow. what is your intention?

Comment (t3easy) [reply .github/workflows/tests.yaml:43]: If someone installs surf as (global) composer package, the dependencies will be installed without respecting the lock only by resolving deplendecies by the semver definitions of the json. Therefore we should test it, to.

Comment (simonschaufi) [reply .github/workflows/tests.yaml:43]: I'm against such checks. That makes only sense before a new release but not for every PR.

Comment (t3easy) [reply .github/workflows/tests.yaml:43]: As a release can be installed at any time with dependencies of that secound, another idea is to run this check scheduled.

Comment (t3easy) [reply .github/workflows/tests.yaml:43]: As the scheduled run is only possible for the default branch, 
> Scheduled workflows run on the latest commit on the default or base branch
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onschedule
and the update only adds 10 secounds, I don't see a time problem.

Comment (simonschaufi) [reply .github/workflows/tests.yaml:43]: what exactly could be the problem when you install surf with the latest dependencies?

Comment (t3easy) [reply .github/workflows/tests.yaml:43]: See last problems of TYPO3 CMS with updated dependency https://forge.typo3.org/issues/91832. Sometimes minor updates ship unexpected breaking changes.
Some also run tests with `--prefer-lowest` to check if the minimum required version with all dependencies is ok. See test matrix of TYPO3 CMS https://github.com/TYPO3/TYPO3.CMS/blob/master/Build/bamboo/src/main/java/core/AbstractCoreSpec.java#L1711-L1766.
They run test against the locked (install), latest update ald lowest dependencies.

Comment (t3easy) [reply .github/workflows/tests.yaml:43]: After all:

- It doesn't hurt.
- We had the tests with travis, too https://github.com/TYPO3/Surf/blob/2.0/.travis.yml#L62-L73
